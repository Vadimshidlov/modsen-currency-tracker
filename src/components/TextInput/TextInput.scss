@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;
@import "@/index.scss";

.currency__input,
.currency__input:disabled {
  @include fonts.PoppinsLight;
  outline: none;
  font-size: fonts.$font-size_20;
  padding: 5px 10px;
  border: 1px solid #ccc;
  border-radius: 6px;
  cursor: pointer;

  @media (max-width: 930px) {
    font-size: fonts.$font-size_18;
  }

  @media (max-width: 768px) {
    font-size: fonts.$font-size_15;
  }

  @media (max-width: 450px) {
    font-size: fonts.$font-size_13;
  }

  @media (max-width: 420px) {
    font-size: fonts.$font-size_12;
  }

  @media (max-width: 320px) {
    font-size: fonts.$font-size_10;
  }
}

.currency__input:disabled {
  opacity: 1;
  background: colors.$color_white;
}
