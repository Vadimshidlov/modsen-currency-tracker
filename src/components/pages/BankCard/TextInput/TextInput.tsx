import React, { Component } from "react";
import { TextInputType } from "@/types/HomePageTypes/types";

class TextInput extends Component<TextInputType> {
    render() {
        const { className, ...rest } = this.props;

        return <input className={className} {...rest} type="text" />;
    }
}

export default TextInput;
