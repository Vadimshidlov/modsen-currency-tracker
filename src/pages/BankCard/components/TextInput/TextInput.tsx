import React, { Component } from "react";
import { TextInputType } from "@/types/HomePageTypes";

class TextInput extends Component<TextInputType> {
    render() {
        const { className, ...rest } = this.props;

        return <input className={className} {...rest} type="text" />;
    }
}

export default TextInput;
