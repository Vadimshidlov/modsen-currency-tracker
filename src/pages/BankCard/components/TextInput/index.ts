export { default } from "@/pages/BankCard/components/TextInput/TextInput";
