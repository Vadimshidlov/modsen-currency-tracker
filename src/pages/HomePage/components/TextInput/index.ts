export { TextInput } from "@/pages/HomePage/components/TextInput/TextInput";
