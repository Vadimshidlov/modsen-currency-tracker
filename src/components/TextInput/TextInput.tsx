import React from "react";
import "@/components/TextInput/TextInput.scss";

export interface ITextInput extends React.ComponentPropsWithRef<"input"> {
    className: string;
}

export function TextInput({ className, ...rest }: ITextInput) {
    return (
        <div>
            <input className={className} {...rest} />
        </div>
    );
}
