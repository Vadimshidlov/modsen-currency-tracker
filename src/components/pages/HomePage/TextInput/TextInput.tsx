import React from "react";
import "@/components/pages/HomePage/TextInput/TextInput.scss";
import { TextInputType } from "@/types/HomePageTypes/types";

export function TextInput({ className, ...rest }: TextInputType) {
    return (
        <div>
            <input className={className} {...rest} data-testid="text-input" />
        </div>
    );
}
