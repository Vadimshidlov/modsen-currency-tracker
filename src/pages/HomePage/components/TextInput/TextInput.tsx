import React from "react";
import "@/pages/HomePage/components/TextInput/TextInput.scss";
import { TextInputType } from "@/types/HomePageTypes";

export function TextInput({ className, ...rest }: TextInputType) {
    return (
        <div>
            <input className={className} {...rest} data-testid="text-input" />
        </div>
    );
}
