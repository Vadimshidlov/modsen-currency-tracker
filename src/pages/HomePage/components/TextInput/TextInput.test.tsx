import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "@/store";
import { renderWithProviders, screen } from "@/utils/rtl-utils";
import { TextInput } from "@/pages/HomePage/components/TextInput/index";

describe("CurrencyStocks component tests", () => {
    it("should correct render component", async () => {
        renderWithProviders(<TextInput className="text-input" placeholder="Hello!" />);

        expect(screen.getByTestId("text-input")).toBeInTheDocument();
        expect(screen.getByTestId("text-input")).toHaveClass("text-input");
    });

    it("snapshot matches", async () => {
        const { asFragment } = renderWithProviders(
            <TextInput className="text-input" placeholder="Hello!" />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <TextInput className="text-input" placeholder="Hello!" />
                </Provider>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
