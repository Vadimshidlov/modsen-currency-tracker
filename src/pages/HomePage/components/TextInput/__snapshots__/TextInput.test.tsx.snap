// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyStocks component tests should match snapshot 1`] = `
<div>
  <input
    className="text-input"
    data-testid="text-input"
    placeholder="Hello!"
  />
</div>
`;

exports[`CurrencyStocks component tests snapshot matches 1`] = `
<DocumentFragment>
  <div>
    <input
      class="text-input"
      data-testid="text-input"
      placeholder="Hello!"
    />
  </div>
</DocumentFragment>
`;
